src/cljs/main/broadfcui/page/workspaces_list.cljs

```diff
@@ -47,21 +50,128 @@
             "Running" [icons/RunningIcon]
             "Exception" [icons/ExceptionIcon]))]))})
 
+(react/defc RequestAuthDomainAccessDialog
+  {:get-initial-state
+   (fn [{:keys [props state]}]
+     {:ws-auth-domains (vec (map
+                             (fn [group]
+                               {:name group
+                                :data {:member? (contains? (:my-auth-domains @state) group)
+                                       :requested? false
+                                       :requesting? false}})
+                             (:ws-auth-domains props)))})
+   :render
+   (fn [{:keys [props state this]}]
+     [comps/OKCancelForm
+      {:header "Request Access"
+       :content
+       (react/create-element
+        (let [{:keys [my-auth-domains ws-instructions error]} @state]
+          (cond
+            (not (or (and my-auth-domains ws-instructions) error))
+            [comps/Spinner {:text "Loading authorization domains..."}]
+            error
+            (case (:code error)
+              (:unknown :parse-error)
+              [:div {:style {:color (:exception-state style/colors)}}
+               "Error:" [:div {} (:details error)]]
+              [comps/ErrorViewer {:error (:details error)}])
+            :else
+            [:div {:style {:width 750}}
+             [:div {} "You cannot access this workspace because it contains restricted data.
+                         You need permission from the admin(s) of all of the Authorization Domains
+                         protecting the workspace."]
+             (let [simple-th (fn [text]
+                               [:th {:style {:textAlign "left" :padding "0 0.5rem"
+                                             :borderBottom style/standard-line}} text])
+                   simple-td (fn [text]
+                               [:td {}
+                                [:label {:style {:display "block" :padding "1rem 0.5rem"
+                                                 :width "33%"}} text]])
+                   instructions (into {}
+                                      (map (fn [ad]
+                                             {(keyword (:groupName ad)) (:instructions ad)})
+                                           ws-instructions))]
```

Comment:
`reduce` might be a little clearer here:
```cljs
(reduce (fn [m ad] (assoc m (keyword (:groupName ad)) (:instructions ad))) {} ws-instructions)
```
The rule of thumb I use: `map` is for turning a collection into another one of the same shape, `reduce` is for changing the shape.